docs/sphinxext/eval.py

```diff
@@ -36,5 +37,36 @@ def run(self):
         return []
 
 
+class PingParameterDirective(Directive):
```

Comment:
Can we reduce the duplication here?  It looks like the only difference is the `get_(ping)?parameter_doc` part.  Maybe make that a method overridden in each class that inherits from a common base class?